src/packages/hex.gleam

```diff
@@ -3,41 +3,204 @@ import gleam/json
 import gleam/http
 import gleam/result
 import gleam/hackney
+import gleam/list
 import gleam/dynamic.{Decoder} as d
+import time/parse.{parse_iso8601_to_gregorian_seconds}
+import gleam/pgo
+import gleam/string
+import gleam/int
+import gleam/option.{Option}
 
 pub type Error {
   HackneyError(hackney.Error)
   JsonError(json.DecodeError)
+  DatabaseError(pgo.QueryError)
+  ItemNotInListError(Nil)
 }
 
-pub fn query() {
+pub fn query(db: pgo.Connection) {
+  try last_scanned = get_last_scanned(db)
+
+  try packages = query_all_packages([], 1, last_scanned)
+
+  // Only update last scanned at if packages were actually
+  // found that way we only ever use the time from
+  // Hex's API 
+  case packages
+  |> list.length >= 1 {
```

Comment:
general tip, counting the length of a list is expensive, especially if you only wanna learn if there's any elements in a list. Pattern matching with `case` does it quickly and also gives you a safe way to assign the first element to a variable.

Typically in Gleam you won't ever use `list.length` unless you are writing `there are X items in this table` in HTML or something like that.

Reply:
👍🏼 I have now fixed this and noted for future.